src/Stratis.Bitcoin/Utilities/IpHelper.cs

```diff
@@ -7,30 +7,37 @@ namespace Stratis.Bitcoin.Utilities
 {
     public static class IpHelper
     {
+        // Don't re-use ports.
+        private static HashSet<uint> usedPorts = new HashSet<uint>();
```

Comment:
Ah interesting idea, you think this will reduce the port in use failures?

Reply:
@quantumagi seeing as this is static, it might be an idea to clear the set whent he node builder starts and on node builder dispose?

Reply:
It may be safer to not re-use ports for the duration of the test run.

Reply:
@dangershony , no not sure, i also have my doubts.

Reply:
I don't know if IpHelper is only used in tests, but it's in Startis.Bitcoin so we should assume it could be used elsewhere and I don't think it's a good idea to use any kind of static stuff in our library.

If we need it, it means we need a singleton service that holds that list and is registered in the DI container.

Reply:
I am happy to remove any static stuff at this point. The simple in-use check here and the retries done elsewhere should be sufficient.